src/main/java/dinkplugin/notifiers/DeathNotifier.java

```diff
@@ -33,57 +42,83 @@ public void onActorDeath(ActorDeath actor) {
         if (isEnabled() && plugin.getClient().getLocalPlayer() == actor.getActor()) {
             this.handleNotify();
         }
+        lastTarget = new WeakReference<>(null);
     }
 
-    private void handleNotify() {
-        Player localPlayer = plugin.getClient().getLocalPlayer();
-        Actor pker = null;
-        for (Player other : plugin.getClient().getPlayers()) {
-            if (other.getInteracting() == localPlayer) {
-                pker = other;
-                break;
-            }
+    public void onInteraction(InteractingChanged event) {
+        if (event.getSource() == plugin.getClient().getLocalPlayer()) {
+            lastTarget = new WeakReference<>(event.getTarget());
         }
-        ItemContainer inventory = plugin.getClient().getItemContainer(InventoryID.INVENTORY);
-        ItemContainer equipment = plugin.getClient().getItemContainer(InventoryID.EQUIPMENT);
+    }
 
-        assert inventory != null;
-        assert equipment != null;
-        List<Pair<Item, Long>> itemsByPrice = Stream.concat(Arrays.stream(inventory.getItems()), Arrays.stream(equipment.getItems()))
-            .map(item -> Pair.of(item, (long) (plugin.getItemManager().getItemPrice(item.getId())) * (long) (item.getQuantity())))
-            .sorted((item1, item2) -> Math.toIntExact(item2.getRight() - item1.getRight()))
-            .collect(Collectors.toList());
+    private void handleNotify() {
+        Player localPlayer = plugin.getClient().getLocalPlayer();
+        Actor pker = identifyPker();
 
-        int keepCount = 3;
-        if (localPlayer.getSkullIcon() != null) {
-            keepCount = 0;
-        }
-        if (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM)) {
-            keepCount += 1;
-        }
-        Long losePrice = itemsByPrice.stream()
+        List<Pair<Item, Long>> itemsByPrice = getPricedItems();
+        int keepCount = (localPlayer.getSkullIcon() == null ? 3 : 0) +
+            (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM) ? 1 : 0);
+        long losePrice = itemsByPrice.stream()
             .skip(keepCount)
             .map(Pair::getRight)
             .reduce(Long::sum)
             .orElse(0L);
 
-        String template = plugin.getConfig().deathNotifyMessage();
-        if (pker != null && plugin.getConfig().deathNotifPvpEnabled()) {
-            template = plugin.getConfig().deathNotifPvpMessage();
-        }
+        String template = pker != null && plugin.getConfig().deathNotifPvpEnabled()
+            ? plugin.getConfig().deathNotifPvpMessage()
+            : plugin.getConfig().deathNotifyMessage();
```

Comment:
Probably technically slightly more efficient but I would argue it's not worth the loss in readability